File: src/main/java/gg/mineral/server/network/packet/play/clientbound/EntityStatusPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.server.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public record EntityStatusPacket(int entityId, byte entityStatus) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(entityId);
        os.writeByte(entityStatus);
    }

    @Override
    public byte getId() {
        return 0x1A;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/EntityStatusPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class EntityStatusPacket(val entityId: Int, val entityStatus: Byte) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeInt(entityId)
        os.writeByte(entityStatus.toInt())
    }

    override val id: Byte
        get() = 0x1A
}
